watch/funcs_test.go

```diff
@@ -68,6 +68,66 @@ func TestKeyWatch(t *testing.T) {
 	}
 }
 
+func TestKeyWatch_With_PrefixDelete(t *testing.T) {
```

Comment:
Note - this is an integration test rather than a unit test, it works against a real consul like other tests in this file